web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/TrialsApiController.java

```diff
@@ -1,242 +1,215 @@
-            .withCredentials(new AWSStaticCredentialsProvider(credentials)).withRegion(s3Region).build();
-
-        S3Object s3object = s3client.getObject("oncokb", "drug-matching/result.json");
-        S3ObjectInputStream inputStream = s3object.getObjectContent();
-        JSONParser jsonParser = new JSONParser();
-        JSONObject jsonObject = (JSONObject) jsonParser.parse(new InputStreamReader(inputStream, "UTF-8"));
-
+        Map<String, Tumor> tumors = ClinicalTrialsUtils
+            .getInstance()
+            .getAllMappingResult();
         Tumor tumor = new Tumor();
-        if (jsonObject.containsKey(oncoTreeCode)) {
-            tumor = getTumor(jsonObject, oncoTreeCode);
+        if (tumors.containsKey(oncoTreeCode)) {
+            tumor = tumors.get(oncoTreeCode);
             if (treatment == null) {
-                return new ResponseEntity<List<Trial>>(tumor.getTrials(), status);
+                return new ResponseEntity<List<ClinicalTrial>>(
+                    ClinicalTrialsUtils
+                        .getInstance()
+                        .getTrialsByIDList(tumor.getTrials()),
+                    status
+                );
             }
-
-            List<Trial> trial = getTrialByTreatment(tumor.getTrials(), treatment);
-            return new ResponseEntity<List<Trial>>(trial, status);
+            List<ClinicalTrial> trial = ClinicalTrialsUtils
+                .getInstance()
+                .filterTrialsByTreatment(
+                    ClinicalTrialsUtils
+                        .getInstance()
+                        .getTrialsByIDList(tumor.getTrials()),
+                    treatment
+                );
+            return new ResponseEntity<List<ClinicalTrial>>(trial, status);
         }
-        return new ResponseEntity<List<Trial>>(new ArrayList<>(), status);
+        return new ResponseEntity<List<ClinicalTrial>>(
+            new ArrayList<>(),
+            status
+        );
     }
 
+    @PublicApi
     @PremiumPublicApi
-    @ApiOperation("Return a list of trials using cancer types")
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "OK", responseContainer = "Map"),
-        @ApiResponse(code = 400, message = "Error", response = String.class)})
-    @RequestMapping(value = "/trials/cancerTypes", produces = {"application/json"}, method = RequestMethod.POST)
-    public ResponseEntity<Map<String, List<Trial>>> trialsGetByCancerTypes(
-        @ApiParam(value = "", required = true) @RequestBody() CancerTypesQuery body)
+    @ApiOperation(
+        "Return a list of clinical trials by cancer type, treatment, location and distance"
+    )
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 200, message = "OK", responseContainer = "Map"),
+            @ApiResponse(
+                code = 400,
+                message = "Error",
+                response = String.class
+            ),
+        }
+    )
+    @RequestMapping(
+        value = "/trials/cancerType",
```

Comment:
The api path does not make sense to me. I would rather use `trials-by-cancer-type`